components/support/nimbus-fml/src/client/inspector.rs

```diff
@@ -0,0 +1,492 @@
+    use crate::client::test_helper::client;
+
+    use super::*;
+
+    #[test]
+    fn test_construction() -> Result<()> {
+        let client = client("./nimbus_features.yaml", "release")?;
+        assert_eq!(
+            client.get_feature_ids(),
+            vec!["dialog-appearance".to_string()]
+        );
+        let f = client.get_feature_inspector("dialog-appearance".to_string());
+        assert!(f.is_some());
+
+        let f = client.get_feature_inspector("not-there".to_string());
+        assert!(f.is_none());
+
+        Ok(())
+    }
+
+    fn error(message: &str, line: u32, col: u32, token: Option<&str>) -> FmlEditorError {
+        FmlEditorError {
+            message: message.to_string(),
+            line,
+            col,
+            hightlight: token.map(str::to_string),
+        }
+    }
+
+    #[test]
+    fn test_get_first_error_invalid_json() -> Result<()> {
+        let client = client("./nimbus_features.yaml", "release")?;
+        let f = client
+            .get_feature_inspector("dialog-appearance".to_string())
+            .unwrap();
+
+        fn test_syntax_error(f: &FmlFeatureInspector, input: &str, col: u32, highlight: bool) {
+            if let Some(e) = f.get_first_error(input.to_string()) {
+                let highlight = if highlight { Some(input) } else { None };
+                assert_eq!(e, error("Need valid JSON object", 0, col, highlight))
+            } else {
+                unreachable!("No error for \"{input}\"");
+            }
+        }
+
+        test_syntax_error(&f, "", 0, false);
+        test_syntax_error(&f, "x", 0, false);
+        test_syntax_error(&f, "{ \"\" }, ", 5, false);
+        test_syntax_error(&f, "{ \"foo\":", 7, false);
+
+        test_syntax_error(&f, "[]", 0, true);
+        test_syntax_error(&f, "1", 0, true);
+        test_syntax_error(&f, "true", 0, true);
+        test_syntax_error(&f, "\"string\"", 0, true);
+
+        assert!(f.get_first_error("{}".to_string()).is_none());
+        Ok(())
+    }
+
+    #[test]
+    fn test_get_first_error_type_invalid() -> Result<()> {
+        let client = client("./nimbus_features.yaml", "release")?;
+        let f = client
+            .get_feature_inspector("dialog-appearance".to_string())
+            .unwrap();
+
+        let s = r#"{}"#;
+        assert!(f.get_first_error(s.to_string()).is_none());
+        let s = r#"{
+            "positive": {}
+        }"#;
+        assert!(f.get_first_error(s.to_string()).is_none());
+
+        let s = r#"{
+            "positive": 1
+        }"#;
+        if let Some(_err) = f.get_first_error(s.to_string()) {
+        } else {
+            unreachable!("No error for \"{s}\"");
+        }
+
+        let s = r#"{
+            "positive1": {}
+        }"#;
+        if let Some(_err) = f.get_first_error(s.to_string()) {
+        } else {
+            unreachable!("No error for \"{s}\"");
+        }
+
+        Ok(())
+    }
+
+    #[test]
+    fn test_find_err() -> Result<()> {
```

Comment:
I'm a little unsure -- is this test covering the case when there's no error to be found? What about if the path is invalid?